bin/thingsvision.py

```diff
@@ -0,0 +1,179 @@
+    type=str,
+    default="cpu",
+    choices=["cpu", "cuda"],
+    help="Device to use for the extractor.",
+)
+
+subparsers = parent_parser.add_subparsers(
+    title="Subcommands",
+    description="Valid subcommands are",
+    help="Additional help available is available within  each subcommand",
+    dest="command",
+)
+
+parser_model = subparsers.add_parser(
+    "show_model", description="Shows the model architecture", parents=[common_parser]
+)
+
+parser_extract = subparsers.add_parser(
+    "extract_features", description="Extract features from images", parents=[common_parser]
+)
+
+parser_extract.add_argument(
+    "--image_root",
+    type=str,
+    help="Path to directory containing images",
+    default="./images",
+)
+parser_extract.add_argument(
+    "--class_names",
+    type=str,
+    help="optional list of class names for class dataset",
+    default=None,
+)
+parser_extract.add_argument(
+    "--file_names",
+    type=str,
+    help="optional list of file names for class dataset",
+    default=None,
+)
+parser_extract.add_argument(
+    "--batch_size", type=int, default=32, help="Batch size used for feature extraction."
+)
+parser_extract.add_argument(
+    "--out_path",
+    type=str,
+    default="./features",
+    help="Path to directory where features should be stored.",
+)
+parser_extract.add_argument(
+    "--model_parameters",
+    type=str,
+    default=None,
+    help="Clip vision model has to be defined in the model_parameters dictionary",
+)
+parser_extract.add_argument(
+    "--module_name",
+    type=str,
+    default="features.10",
+    help="Name of the module to use for feature extraction.",
+)
+parser_extract.add_argument(
+    "--file_format",
+    type=str,
+    default="npy",
+    help="File format to use for storing features.",
+    choices=["npy", "hdf5", "mat", "txt"],
+)
+parent_parser.add_argument(
+    "-h",
+    "--help",
+    action="help",
+    default=argparse.SUPPRESS,
+    help="Show this help message and exit.",
+)
+parent_parser.add_argument(
+    "-v",
+    "--version",
+    action="version",
+    version="%(prog)s 2.2.8",
+    help="Show program's version number and exit.",
+)
+
+
+def main():
+    args = parent_parser.parse_args()
+    if len(sys.argv) == 1:
+        parent_parser.print_help(sys.stderr)
+        sys.exit(1)
+    elif len(sys.argv) == 2:
+        print("\nPlease specify optional commands and arguments:\n")
+        if args.command == "show_model":
+            parser_model.print_help(sys.stderr)
+        elif args.command == "extract_features":
+            parser_extract.print_help(sys.stderr)
+        sys.exit(1)
+    try:
+        from thingsvision import get_extractor
+        from thingsvision.utils.data import ImageDataset, DataLoader
+        from thingsvision.utils.storing import save_features
+    except ImportError:
+        print(
+            "Thingsvision is not installed. Please install it using pip install thingsvision"
+        )
+        sys.exit(1)
```

Comment:
This seems odd to me. How can the CLI work if thingsvision is not installed? If it can, then this is quite unexpected behavior which is something we want to avoid. The CLI should work if and only if thingsvision is installed.